lib/auth0/mixins/validation.rb

```diff
@@ -66,20 +72,27 @@ def validate_signature(decoded_id_token)
             raise Auth0::InvalidIdToken, 'Invalid ID token signature'
           rescue JWT::IncorrectAlgorithm
             alg = decoded_id_token[:header]['alg']
-            raise Auth0::InvalidIdToken, "Signature algorithm of \"#{alg}\" is not supported. Expected the ID token to be signed with \"#{algorithm_name}\""
+            raise Auth0::InvalidIdToken, "Signature algorithm of \"#{alg}\" is not supported. Expected the ID token to be signed with \"#{algorithm.name}\""
           rescue JWT::DecodeError
             kid = decoded_id_token[:header]['kid']
             raise Auth0::InvalidIdToken, "Could not find a public key for Key ID (kid) \"#{kid}\""
           end
         end
+        # rubocop:enable Metrics/MethodLength, Metrics/AbcSize
 
-        def validate_claims(decoded_id_token); end
+        def validate_claims(decoded_id_token); end # TODO: Implement (in the next PR)
+      end
+
+      class JWTAlgorithm
+        def name
+          self.class.to_s.split('::').last
```

Comment:
This is wrong. Use a string to define the name of the class in a class property.

Reply:
Changed in 64bc3aa057870bb29267f94fd0333c74b958ed82